contracts/Delay.sol

```diff
@@ -80,90 +136,198 @@ contract Delay is AragonApp, IForwarder {
+            _closeAgreementAction(delayedScript.actionId);
+        }
+
+        emit ExecutedScript(_delayedScriptId, delayedScript.actionId);
+        _deleteDelayedScript(_delayedScriptId);
+    }
+
+    /**
+    * @notice Close agreement action with ID `_delayedScriptId`
+    * @param _delayedScriptId The ID of the script to close the agreement action of
+    */
+    function closeAgreementAction(uint256 _delayedScriptId) external {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+
+        require(_canExecute(delayedScript), ERROR_CAN_NOT_EXECUTE);
+        require(delayedScript.disputableStatus != DisputableStatus.Closed, ERROR_ACTION_CLOSED);
+
+        _closeAgreementAction(delayedScript.actionId);
+        emit AgreementActionClosed(_delayedScriptId, delayedScript.actionId);
+
+        delayedScript.disputableStatus = DisputableStatus.Closed;
     }
 
     /**
     * @notice Return whether a script with ID #`_scriptId` can be executed
-    * @param _scriptId The ID of the script to execute
+    * @param _delayedScriptId The ID of the script to execute
     */
-    function canExecute(uint256 _scriptId) public view returns (bool) {
-        bool withinExecutionWindow = getTimestamp64() > delayedScripts[_scriptId].executionTime;
-        bool isUnpaused = !_isExecutionPaused(_scriptId);
+    function canExecute(uint256 _delayedScriptId) public view returns (bool) {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+        return _canExecute(delayedScript);
+    }
 
-        return withinExecutionWindow && isUnpaused;
+    /**
+    * @dev IForwarder interface conformance
+    */
+    function isForwarder() external pure returns (bool) {
+        return true;
     }
 
+    /**
+    * @dev IForwarder interface conformance
+    */
     function canForward(address _sender, bytes) public view returns (bool) {
         return canPerform(_sender, DELAY_EXECUTION_ROLE, arr());
     }
 
     /**
     * @notice Delays execution for `@transformTime(self.executionDelay(): uint)`
+    * @dev IForwarder interface conformance
     * @param _evmCallScript The script that can be executed after a delay
     */
     function forward(bytes _evmCallScript) public {
         require(canForward(msg.sender, _evmCallScript), ERROR_CAN_NOT_FORWARD);
-        _delayExecution(_evmCallScript);
+        _delayExecution(new bytes(0), _evmCallScript);
+    }
+
+    /**
+    * @dev Challenge script execution, IDisputable interface conformance
+    * @param _delayedScriptId The ID of the script execution to challenge
+    */
+    function _onDisputableActionChallenged(uint256 _delayedScriptId, uint256 /* _challengeId */, address /* _challenger */)
+        internal
+    {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+        delayedScript.disputableStatus = DisputableStatus.Challenged;
+        _pauseExecution(delayedScript);
```

Comment:
Careful that you are updating the state before pausing the action

Reply:
The DisputableStatus is independent of the action status so the order doesn't matter. However now you mention it I can probably unify them.